src/Rings/FreeAssAlgIdeal.jl

```diff
@@ -17,18 +17,24 @@ Two-sided ideal of a free associative algebra with elements of type `T`.
 mutable struct FreeAssAlgIdeal{T} <: Ideal{T}
   gens::IdealGens{T}
   gb::IdealGens{T}
-
-  function FreeAssAlgIdeal(R::FreeAssAlgebra, g::Vector{T}) where T <: FreeAssAlgElem
+  deg_bound::Int
+  function FreeAssAlgIdeal(g::IdealGens{T}, check_groebner_basis::Bool=false) where T <: FreeAssAlgElem
     r = new{T}()
-    r.gens = IdealGens(R, g)
+    r.gens = g
+    @req !check_groebner_basis || base_ring(base_ring(g)) isa Field "Groebner basis check requires a field base ring"
+    if check_groebner_basis && base_ring(base_ring(g)) isa Field && is_groebner_basis(g)
+      r.gb = g
+      r.deg_bound = -1
+    end
     return r
   end
 end
 
-function AbstractAlgebra.expressify(a::FreeAssAlgIdeal; context = nothing)
-  return Expr(:call, :ideal, [expressify(g, context = context) for g in collect(a.gens)]...)
+Base.show(io::IO, a::FreeAssAlgIdeal) = print(io, "Ideal of ", base_ring(a), " with ", number_of_generators(a), " generators")
+
+function Base.:(==)(I2::FreeAssAlgIdeal, I1::FreeAssAlgIdeal)
```

Comment:
every type that implements `==` must also implement `hash(::T, ::UInt)` to allow for correctness with julia Base (see e.g. https://github.com/oscar-system/Oscar.jl/issues/2222).
As there is no normal form for these ideals, one needs to use the limited information one has to construct a really bad (but valid) hash function, e.g. as
```julia
function Base.hash(I::FreeAssAlgIdeal, h::UInt)
  return hash(base_ring(I), h)
end
```